main.go

```diff
@@ -268,18 +268,36 @@ func parseURIList(uriList []string, logger *logrus.Logger, splitCluster bool) []
 	return URIs
 }
 
+// buildURIManually builds the URI manually by checking if the user and password are supplied
+func buildURIManually(uri string, user string, password string) string {
+	uriArray := strings.SplitN(uri, "://", 2) //nolint:mnd
+	prefix := uriArray[0] + "://"
+	uri = uriArray[1]
+
+	// IF user@pass not contained in uri AND custom user and pass supplied in arguments
+	// DO concat a new uri with user and pass arguments value
+	if !strings.Contains(uri, "@") && user != "" && password != "" {
```

Comment:
I'm wondering if we shouldn't support a case where the user is non-empty and the password is empty.

Reply:
good point for improvement, but in scope of this fix I think we can leave as it used to be. I just reverted old logic for cases where url.Parse doesn't work. Additionally for that users can set user in initial DSN. let's create github issue with good-first-task label for those who would like to contribute